pkg/constants/constants.go

```diff
@@ -566,3 +570,13 @@ const (
 func GetMergedPullSecretName(cd *hivev1.ClusterDeployment) string {
 	return apihelpers.GetResourceName(cd.Name, mergedPullSecretSuffix)
 }
+
+// GetImagePullSecretName returns name for image pull secret name per cluster deployment
+func GetImagePullSecretName(cd *hivev1.ClusterDeployment) string {
+	return apihelpers.GetResourceName(cd.Name, hiveImagePullSecretSuffix)
+}
+
+// GetImagePullSecretNameForDeprovision returns name for image pull secret name per cluster deprovision
+func GetImagePullSecretNameForDeprovision(cd *hivev1.ClusterDeprovision) string {
+	return apihelpers.GetResourceName(cd.Name, hiveImagePullSecretSuffix)
+}
```

Comment:
I think we can DRY these by making the param a meta.Object and using GetName() on it.

(Moot if you copy the Secret down just once with its original name, as noted above.)